src/hermes_types.h

```diff
@@ -42,7 +42,6 @@ typedef double f64;
 typedef u16 DeviceID;
 
 namespace api {
-typedef int Status;
```

Comment:
Is there a reason you moved the `Status` from the `hermes::api` namespace to the `hermes` namespace? In my opinion, every object that a Hermes-native user will interact with belongs in `hermes::api` and `src/api`.

Reply:
> Can we remove the `try ... catch` blocks from the tests? An uncaught exception will print the `what()` automatically, so catching just to print `what()` is redundant.

That's true. Most tests would be fine without `try ... catch`, but the mdm test will fail. Do you suggest we modify it? 
Function [TestMaxNameLength](https://github.com/HDFGroup/hermes/blob/kimmy/error_handling/test/mdm_test.cc#L168) is testing if bucket [`invalid_bucket`](https://github.com/HDFGroup/hermes/blob/kimmy/error_handling/test/mdm_test.cc#L173) can be created if the name is too long, if `throw` in the constructor is not caught in time, the test will fail. This will happen to [blob name test](https://github.com/HDFGroup/hermes/blob/kimmy/error_handling/test/mdm_test.cc#L185) as well.

Reply:
The reason is that `Status` is used by functions that belong to `src` as well, not just `src/api`. But it is true that the user will only interact with `hermes::api`.

Reply:
For the convenience of users, I will move it to `hermes::api`.